### ISSUE TITLE
React 17

### ISSUE THREAD

**Issue by nazstone** [react-native-maps/react-native-maps#3812]
Hello,

I am using react-native in version 0.64 with react 17. This version is not compatible because of the peerDependencies. Do you know if it will evolve to the new version?

Thanks.

**Comment by muresan-roland** [comment]
Hey, 

I have the same issue, can't install on a app created on the latest version of react and react native.

**Comment by vallesluiggi** [comment]
Me too, please, we need help :-D

**Comment by vallesluiggi** [comment]
> Me too, please, we need help :-D

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: project@0.0.1
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"17.0.1" from the root project
npm ERR!   peer react@">= 16.0 || < 17.0" from react-native-maps@0.28.0
npm ERR!   node_modules/react-native-maps
npm ERR!     react-native-maps@"*" from the root project
npm ERR!   2 more (react-native, react-native-web)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"17.0.2" from react-dom@17.0.2
npm ERR! node_modules/react-dom
npm ERR!   peer react-dom@">=16.5.1" from react-native-web@0.11.7
npm ERR!   node_modules/react-native-web
npm ERR!     peer react-native-web@"^0.11" from react-native-maps@0.28.0
npm ERR!     node_modules/react-native-maps
npm ERR!       react-native-maps@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

**Comment by alexandrtovmach** [comment]
Works fine for me with next config:
```
"react": "17.0.2",
"react-native": "0.64.1",
```

**Comment by nazstone** [comment]
> Works fine for me with next config:
> 
> ```
> "react": "17.0.2",
> "react-native": "0.64.1",
> "react-native-maps": "^0.28.0",
> ```

same issue:
It does not for me with 
```
"react": "17.0.2",
"react-native": "0.64.1",
"react-native-maps": "^0.28.0",
```
    
```
npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: nazinvader@0.0.1
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"17.0.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"17.0.1" from react-native@0.64.1
npm ERR! node_modules/react-native
npm ERR!   react-native@"0.64.1" from the root project
```

Do you force install?

**Comment by alexandrtovmach** [comment]
@nazstone Just for testing purpose, could you try to install packages with `yarn`?

**Comment by nazstone** [comment]
Hi @alexandrtovmach, using `yarn` work fine.

I try also to remove node_modules, package-lock.json and yarn.lock but this doesnt work. I am pretty sure it is link to the peerDependencies from the package.json of `react-native-maps' library.

So I will now use yarn. I tell you if something wrong happened using the lib.

**Comment by alexandrtovmach** [comment]
@nazstone yes, yarn resolving packages in a different way than npm, that's why it works. So you can use it as workaround.

Anyway if you interested in fixing the issue to continue using npm, just put things that I asked, and I'll try to help.

**Comment by bewannabe96** [comment]
This is a hack which is not recommended if you don't know what you are doing.
So I just manually cloned and installed myself.

1. clone repo into /node_modules
`cd node_modules && git clone https://github.com/react-native-maps/react-native-maps.git`
2. config `Podfile`
add `pod 'react-native-maps', :path => '../node_modules/react-native-maps'` into `Podfile`
3. `cd ios && pod install`

This way you can use the package like normal at least on IOS.
(I only needed IOS version so haven't configured Android part)

**Comment by bewannabe96** [comment]
just force install package 

`npm install react-native-maps --force`

**Comment by nazstone** [comment]
> @nazstone yes, yarn resolving packages in a different way than npm, that's why it works. So you can use it as workaround.
> 
> Anyway if you interested in fixing the issue to continue using npm, just put things that I asked, and I'll try to help.

Thanks I will use yarn.

**Comment by vallesluiggi** [comment]
Hello, every all.

The steps I gave to make it work:
Update to version 17.0.2 of ReactJS
Remove Package-lock.json, yarn.lock, node_modules
Change in the package.json the version of the reactjs manually 17.0.2
-yarn upgrade --dev

yarn add react-native-maps -E
-ncu -u
There the map was installed correctly, but I had some other things that I could not install with yarn and therefore it gave me some problems, so I executed:
npm install

and voila I have the maps already working in the latest version of ReactJS with React Native

![image](https://user-images.githubusercontent.com/19999481/117577773-3a63f600-b0b9-11eb-89ab-aae44daa7e6b.png)

Regards,

**Comment by fiznool** [comment]
I'm not quite sure why this was closed. Force installing is not a sensible option as it will cause any other invalid peer dependencies inside your dependency resolution chain to also be installed.

The real fix here is to change the peer dependency resolution field in the `package.json` for this project to `"react": "16.x || 17.x"`. Otherwise, people using npm 7 and React Native 0.64 will see this error and will not be able tom proceed without force-installing.

The reason that some people are seeing this error, and others aren't: npm v7 has started installing peer dependencies automatically. If two of your dependencies ask for the same peer dependency package, but a different version, this is a package mismatch and npm v7 will bail out when installing. In this case, because React Native 0.64 has a peer dependency of React 17, and this repo has a peer dependency of React 16, npm can't continue. If you are on npm v6, or yarn, you won't see this error.

More info: https://github.com/npm/rfcs/blob/latest/implemented/0025-install-peer-deps.md
